src/vs/platform/remote/common/remoteExtensionsScanner.ts

```diff
@@ -13,6 +15,9 @@ export const RemoteExtensionsScannerChannelName = 'remoteExtensionsScanner';
 export interface IRemoteExtensionsScannerService {
 	readonly _serviceBrand: undefined;
 
-	whenExtensionsReady(): Promise<void>;
+	/**
+	 * Returns a promise that resolves to an array of extension identifiers or URIs that failed to install
+	 */
+	whenExtensionsReady(): Promise<{ extensions: Array<string | URI>; installOptions: InstallOptions }>;
```

Comment:
`{ extensions: Array<string | URI>; installOptions: InstallOptions }` extract this into a type.

Reply:
I would change the type to following to be more explicit

```
interface InstallExtensionsResult {
	failed: {
		extension: string | URI;
		installOptions: InstallOptions
	}[];
}
```

Reply:
Gotcha.  I will have to take a closer look, but it didn't seem like there was a way for `installOptions` to end up varied per extension - it is built from command line parameters and cannot be changed per extension

```
	const installOptions: InstallOptions = {
				isMachineScoped: !!environmentService.args['do-not-sync'],
				installPreReleaseVersion: !!environmentService.args['pre-release'],
				isApplicationScoped: true // extensions installed during server startup are available to all profiles
			};
```

In that case it feels redundant to store that information for each extension separately.

Reply:
changed in https://github.com/microsoft/vscode/pull/238815/commits/151dd09ab56f0af66e77f70d322b163208771aa8